[Header] Header 컴포넌트 1차 PR 내용 반영

dani-day opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

**1차 PR 리뷰사항 반영**
- width 375 이하 시, 깨지는 현상 수정하였습니다.
- sitemap 바깥 blur 처리하였으며, scroll 방지 완료하였습니다.
- 아코디언 토글의 화살표 및 버튼의 마크업, 레이아웃을 조정하였습니다.
- 모바일 header 는 높이를 기존의 50%로 조정하였습니다.
- 추후 기능 고려를 위한 ENG 버튼 (기능 없음) 을 추가하였습니다.
- 배포 시 이미지 경로 이슈 해결하였습니다. (import 로 불러옴)

https://github.com/ReOpenSG/ReOpenSG/assets/134567470/49e95523-a324-4bc7-9d4f-7c4d1ad884bb


## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 1차 PR 리뷰사항 반영

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- 없음

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

dani-day committed:
🐛 [Fix] : 1차 PR 리뷰사항 전체 반영 (375 이하 고려, sitemap 바깥 blur 처리, sitemap 바깥 스크롤 방지, 화살표 및 버튼 레이아웃 조정)

+ width 375 이하 시, 깨지는 현상 수정하였습니다.
+ sitemap 바깥 blur 처리하였으며, scroll 방지 완료하였습니다.
+ 아코디언 토글의 화살표 및 버튼의 마크업, 레이아웃을 조정하였습니다.
+ 모바일 header 는 높이를 기존의 50%로 조정하였습니다.
+ 추후 기능 고려를 위한 ENG 버튼을 추가하였습니다.

Related to: #34

dani-day committed:
🐛 [Fix] : Header 컴포넌트 내 이미지/아이콘 경로 이슈 해결

+ Header 컴포넌트 내 이미지/아이콘 경로 이슈 해결하였습니다.

Related to: #34

ingbinsee commented on src/layout/Header.jsx:
```diff
@@ -179,18 +204,22 @@ function Header() {
+        <li className="flex justify-center items-center px-open-md py-open-sm text-open-font-small desktop:text-open-font-medium tablet:text-open-font-medium">
+          <div className="flex flex-row gap-open-md">
+            <button className={activeLanguage !== 'ENG' ? '-text--openfoundation-secondary' : '-text--openfoundation-tertiary'} type="button" onClick={() => handleLanguageActive('KOR')}>KOR</button>
+            <div className="w-[1px] -bg--openfoundation-secondary" />
```
P3: 해당 부분은 구분선이므로 ```-bg--openfoundation-secondary``` 대신 ```-bg--openfoundation-tertiary```으로 색상을 설정하면 더 좋을 듯 합니다. 또한, div가 아닌 KOR과 ENG 사이의 border로 설정해도 좋을 것 같습니다.

dani-day replied:
`-bg--openfoundation-secondary` 대신 -`bg--openfoundation-tertiary` 으로 구분선 색상 변경하였습니다 :)

![스크린샷 2023-12-15 오후 2 49 43](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/a6530712-6218-4d2d-a0d2-54bf68885e45)

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다.

- Header.jsx(P3 review)

dani-day committed:
🌈 [Design] : 언어 구분선 변경 (-bg--openfoundation-secondary -> -bg--openfoundation-tertiary)

+ 언어 구분선 색상을 변경하였습니다.

Related to: #65

dani-day committed:
♻️ [Refactor] : boolean 값에 따라 랜더링이 정해지는 컴포넌트를 prop 만 전달하는 방식으로 변경

+ AccordionItem 컴포넌트에서 SitemapArrow 의 방향을 정해주는 boolean 상태를 prop 으로 전달하는 방식으로 변경

Related to:  #61

kelvin-chihyun commented on src/components/Header/Sitemap.jsx:
P3: hover시 css 변화가 있었으면 좋겠습니다!

dani-day replied:
의견 감사합니다. 추후 반영 고려하겠습니다 :)

kelvin-chihyun commented on src/components/Header/SitemapMobile.jsx:
P3: hover시 css 변화가 있었으면 좋겠습니다!

dani-day replied:
의견 감사합니다. 추후 반영 고려하겠습니다 :)

kelvin-chihyun reviewed (approved):
남긴 의견 외에 이상 없어보입니다. 고생하셨습니다.
